sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobAsyncClient.java

```diff
@@ -591,45 +607,122 @@ Mono<EncryptedBlob> encryptBlob(Flux<ByteBuffer> plainTextFlux) {
                 .map(encryptedKey -> {
                     WrappedKey wrappedKey = new WrappedKey(keyId, encryptedKey, keyWrapAlgorithm);
 
-                    // Build EncryptionData
-                    EncryptionData encryptionData = new EncryptionData()
-                        .setEncryptionMode(CryptographyConstants.ENCRYPTION_MODE)
-                        .setEncryptionAgent(new EncryptionAgent(CryptographyConstants.ENCRYPTION_PROTOCOL_V1,
-                            EncryptionAlgorithm.AES_CBC_256))
-                        .setKeyWrappingMetadata(keyWrappingMetadata)
-                        .setContentEncryptionIV(cipher.getIV())
-                        .setWrappedContentKey(wrappedKey);
-
-                    // Encrypt plain text with content encryption key
-                    Flux<ByteBuffer> encryptedTextFlux = plainTextFlux.map(plainTextBuffer -> {
-                        int outputSize = cipher.getOutputSize(plainTextBuffer.remaining());
-
-                        /*
-                         * This should be the only place we allocate memory in encryptBlob(). Although there is an
-                         * overload that can encrypt in place that would save allocations, we do not want to overwrite
-                         * customer's memory, so we must allocate our own memory. If memory usage becomes unreasonable,
-                         * we should implement pooling.
-                         */
-                        ByteBuffer encryptedTextBuffer = ByteBuffer.allocate(outputSize);
-
-                        int encryptedBytes;
-                        try {
-                            encryptedBytes = cipher.update(plainTextBuffer, encryptedTextBuffer);
-                        } catch (ShortBufferException e) {
-                            throw LOGGER.logExceptionAsError(Exceptions.propagate(e));
-                        }
-                        encryptedTextBuffer.position(0);
-                        encryptedTextBuffer.limit(encryptedBytes);
-                        return encryptedTextBuffer;
-                    });
-
-                    /*
-                     * Defer() ensures the contained code is not executed until the Flux is subscribed to, in
-                     * other words, cipher.doFinal() will not be called until the plainTextFlux has completed
-                     * and therefore all other data has been encrypted.
-                     */
-                    encryptedTextFlux = Flux.concat(encryptedTextFlux,
-                        Mono.fromCallable(() -> ByteBuffer.wrap(cipher.doFinal())));
+                    EncryptionData encryptionData;
+                    Flux<ByteBuffer> encryptedTextFlux;
+                    switch (this.encryptionVersion) {
+                        case V1:
```

Comment:
Similar comment as before on making the large switch cases into method calls to help readability